PowerEditor/src/WinControls/ContextMenu/ContextMenu.cpp

```diff
@@ -115,12 +115,16 @@ void ContextMenu::create(HWND hParent, const std::vector<MenuItemUnit> & menuIte
 
 		if (mainMenuHandle)
 		{
-			bool isEnabled = (::GetMenuState(mainMenuHandle, item._cmdID, MF_BYCOMMAND)&(MF_DISABLED|MF_GRAYED)) == 0;
-			bool isChecked = (::GetMenuState(mainMenuHandle, item._cmdID, MF_BYCOMMAND)&(MF_CHECKED)) != 0;
-			if (!isEnabled)
-				enableItem(item._cmdID, isEnabled);
-			if (isChecked)
-				checkItem(item._cmdID, isChecked);
+			UINT s = ::GetMenuState(mainMenuHandle, item._cmdID, MF_BYCOMMAND);
+			if (s != -1)

```

Comment:
I know MS says:
```
Return value
Type: UINT

If the specified item does not exist, the return value is -1.
...
```
But I thought it would generate warning on compiling, since you're comparing between an unsigned value and a signed one. Or is the flat earth plot true ?

Reply:
Microsoft states here: https://docs.microsoft.com/en-us/cpp/cpp/type-conversions-and-type-safety-modern-cpp?view=msvc-160

> The compiler does not warn about implicit conversions between signed and unsigned integral types. Therefore, we recommend that you avoid signed-to-unsigned conversions altogether. 

Nice recommendation, but hard to follow. It's easier to live on the flat earth.

I could
- add `(UINT)` before `-1`
- change the type of `s` to `int`
- leave it as it is

All variants produce the same code. I tend to leave it.

In this moment, I see that I actually forgot to use the new variable `s`. So I will revise this PR and remove the 2 unnecessary calls of `GetMenuState`.